app/db.js

```diff
@@ -7,42 +7,48 @@ const nconf = require('../config');
 const dbUrl = nconf.get('db');
 mongoose.Promise = Promise;
 
+let tearingDown = false;
+
 const connect = function () {
-  /**
-   const options = {
-    server: {
-      socketOptions: {keepAlive: 1},
-      auto_reconnect: true
-    }
-  }; */
   const options = {
     useMongoClient: true,
-    // logger: logger,
-    loggerLevel: 'warning' // @todo fetch from config file
+    logger: logger.log.bind(logger),
+    loggerLevel: 'debug' // @todo fetch from config file
   };
-  logger.info(`Create MongoDB connection: ${dbUrl}`);
+
+  // Resolve after database connection is ready and listening
+  const connectionPending = new Promise((resolve) => {
+    mongoose.connection.once('connected', resolve);
```

Comment:
if this doesn't happen ever we kind of lose this promise..

Reply:
Well it needs to happen at some point or there is not much we can do, maybe we could use Bluebirds promise timeout to automatically reject it if it doesn't resolve in acceptable time?

Reply:
it still doesn't stop listening event and then it calls already solved promises..